### ISSUE TITLE
Deployment fails on Windows 8.1 with VirtualBox 4.3.8 via Git Bash prompt

### ISSUE THREAD

**Issue by IanHopkinson** [alphagov/pp-development#36]
I'm trying to bring up the development environment on Windows 8.1 with Virtualbox 4.3.8. I had the following log after the `vagrant up` command on the Git Bash prompt. I switched to the virtualbox-4-3-6 branch before starting the process.  At the very end I have copied the complete log, prior to that I have copied only those parts that appear in red.

I appreciate the importance of the version mismatch for Guest Additions but I thought a couple of minor revision points would not be enough to break, and it would be worth knowing if it does break for this reason.

The first "red" error I get is:
`stdin: is not a tty`
This is followed by other output then:

```
/var/lib/gems/1.9.1/gems/bundler-1.5.3/lib/bundler/runtime.rb:220: warning: Insecure world writable dir /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/bin in PATH
, mode 040777
/var/lib/gems/1.9.1/gems/bundler-1.5.3/lib/bundler/runtime.rb:220: warning: Insecure world writable dir /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/bin in PATH
, mode 040777
Error: Operation not permitted - /var/apps/pp-puppet/.tmp/librarian/cache/source/puppet/forge/3792e516e3ff92a0ef9f5e827f8e76eb/puppetlabs/stdlib/version/804879a
f3577ffc462dba2bf74cceb6b/stdlib/spec/unit/puppet/parser/functions/is_domain_name_spec.rb
Error: Try 'puppet help module install' for usage
/var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-puppet-0.9.13/lib/librarian/puppet/source/forge.rb:114:in `unlink': Directory not empty - /var/apps/
pp-puppet/.tmp/librarian/cache/source/puppet/forge/3792e516e3ff92a0ef9f5e827f8e76eb/puppetlabs/stdlib/version/804879af3577ffc462dba2bf74cceb6b (Errno::ENOTEMPTY
)
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-puppet-0.9.13/lib/librarian/puppet/source/forge.rb:114:in `unlink'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-puppet-0.9.13/lib/librarian/puppet/source/forge.rb:114:in `cache_version_unpacked!'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-puppet-0.9.13/lib/librarian/puppet/source/forge.rb:57:in `install_version!'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-puppet-0.9.13/lib/librarian/puppet/source/forge.rb:263:in `install!'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-0.1.2/lib/librarian/manifest.rb:101:in `install!'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-0.1.2/lib/librarian/action/install.rb:49:in `block in install_manifests'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-0.1.2/lib/librarian/action/install.rb:48:in `each'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-0.1.2/lib/librarian/action/install.rb:48:in `install_manifests'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-0.1.2/lib/librarian/action/install.rb:39:in `perform_installation'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-0.1.2/lib/librarian/action/install.rb:12:in `run'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-0.1.2/lib/librarian/cli.rb:165:in `install!'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-puppet-0.9.13/lib/librarian/puppet/cli.rb:71:in `install'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/thor-0.18.1/lib/thor/command.rb:27:in `run'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/thor-0.18.1/lib/thor/invocation.rb:120:in `invoke_command'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/thor-0.18.1/lib/thor.rb:363:in `dispatch'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/thor-0.18.1/lib/thor/base.rb:439:in `start'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-0.1.2/lib/librarian/cli.rb:26:in `block (2 levels) in bin!'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-0.1.2/lib/librarian/cli.rb:31:in `returning_status'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-0.1.2/lib/librarian/cli.rb:26:in `block in bin!'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-0.1.2/lib/librarian/cli.rb:47:in `with_environment'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-0.1.2/lib/librarian/cli.rb:26:in `bin!'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-puppet-0.9.13/bin/librarian-puppet:7:in `<top (required)>'
        from ./bin/librarian-puppet:12:in `load'
        from ./bin/librarian-puppet:12:in `<main>'
The following SSH command responded with a non-zero exit status.
Vagrant assumes that this means the command failed!

chmod +x /tmp/vagrant-shell && /tmp/vagrant-shell

Stdout from the command:
```

There is some further output followed by:

```
Error: Operation not permitted - /var/apps/pp-puppet/.tmp/librarian/cache/source/puppet/forge/3792e516e3ff92a0ef9f5e827f8e76eb/puppetlabs/stdlib/version/804879a
f3577ffc462dba2bf74cceb6b/stdlib/spec/unit/puppet/parser/functions/is_domain_name_spec.rb
Error: Try 'puppet help module install' for usage
```

Full output:

```
Bringing machine 'pp-development-1' up with 'virtualbox' provider...
[pp-development-1] Box 'pp-ubuntu-12-04-vagrant-20140210-virtualbox436' was not found. Fetching box from specified URL for
the provider 'virtualbox'. Note that if the URL does not have
a box for this provider, you should interrupt Vagrant now and add
the box yourself. Otherwise Vagrant will attempt to download the
full box prior to discovering this error.
Downloading box from URL: https://s3-eu-west-1.amazonaws.com/gds-boxes/pp-ubuntu-12-04-vagrant-20140210.box
Extracting box...ate: 2665k/s, Estimated time remaining: --:--:--)
Successfully added box 'pp-ubuntu-12-04-vagrant-20140210-virtualbox436' with provider 'virtualbox'!
[pp-development-1] Importing base box 'pp-ubuntu-12-04-vagrant-20140210-virtualbox436'...
[pp-development-1] Matching MAC address for NAT networking...
[pp-development-1] Setting the name of the VM...
[pp-development-1] Clearing any previously set forwarded ports...
[pp-development-1] Clearing any previously set network interfaces...
[pp-development-1] Preparing network interfaces based on configuration...
[pp-development-1] Forwarding ports...
[pp-development-1] -- 22 => 2222 (adapter 1)
[pp-development-1] Running 'pre-boot' VM customizations...
[pp-development-1] Booting VM...
[pp-development-1] Waiting for machine to boot. This may take a few minutes...
DL is deprecated, please use Fiddle
[pp-development-1] Machine booted and ready!
[pp-development-1] Setting hostname...
[pp-development-1] Configuring and enabling network interfaces...
[pp-development-1] Mounting shared folders...
[pp-development-1] -- /vagrant
[pp-development-1] -- /var/apps
[pp-development-1] Running provisioner: shell...
[pp-development-1] Running: C:/Users/Ian/AppData/Local/Temp/vagrant-shell20140303-808-l9f2rj
stdin: is not a tty
[bootstrap] --> Set root profile to quieten sudo tty message
[bootstrap] --> Set nameserver to 8.8.8.8
[bootstrap] --> Set sources.list to gb and update apt cache
[bootstrap] dpkg-preconfigure: Unable to re-open stdin: No such file or directory
[bootstrap] Selecting previously unselected package libcurl3.
[bootstrap] (Reading database ... 53042 files and directories currently installed.)
[bootstrap] Unpacking libcurl3 (from .../libcurl3_7.22.0-3ubuntu4.7_amd64.deb) ...
[bootstrap] Selecting previously unselected package curl.
[bootstrap] Unpacking curl (from .../curl_7.22.0-3ubuntu4.7_amd64.deb) ...
[bootstrap] Selecting previously unselected package liberror-perl.
[bootstrap] Unpacking liberror-perl (from .../liberror-perl_0.17-1_all.deb) ...
[bootstrap] Selecting previously unselected package git-man.
[bootstrap] Unpacking git-man (from .../git-man_1%3a1.7.9.5-1_all.deb) ...
[bootstrap] Selecting previously unselected package git.
[bootstrap] Unpacking git (from .../git_1%3a1.7.9.5-1_amd64.deb) ...
[bootstrap] Selecting previously unselected package unattended-upgrades.
[bootstrap] Unpacking unattended-upgrades (from .../unattended-upgrades_0.76ubuntu1_all.deb) ...
[bootstrap] Selecting previously unselected package python-pycurl.
[bootstrap] Unpacking python-pycurl (from .../python-pycurl_7.19.0-4ubuntu3_amd64.deb) ...
[bootstrap] Selecting previously unselected package python-software-properties.
[bootstrap] Unpacking python-software-properties (from .../python-software-properties_0.82.7.7_all.deb) ...
[bootstrap] Processing triggers for man-db ...
[bootstrap] Processing triggers for ureadahead ...
[bootstrap] Setting up libcurl3 (7.22.0-3ubuntu4.7) ...
[bootstrap] Setting up curl (7.22.0-3ubuntu4.7) ...
[bootstrap] Setting up liberror-perl (0.17-1) ...
[bootstrap] Setting up git-man (1:1.7.9.5-1) ...
[bootstrap] Setting up git (1:1.7.9.5-1) ...
[bootstrap] Setting up unattended-upgrades (0.76ubuntu1) ...
[bootstrap] Setting up python-pycurl (7.19.0-4ubuntu3) ...
[bootstrap] Setting up python-software-properties (0.82.7.7) ...
[bootstrap] Processing triggers for libc-bin ...
[bootstrap] ldconfig deferred processing now taking place
[bootstrap] gpg: keyring `/tmp/tmpwMo351/secring.gpg' created
[bootstrap] gpg: keyring `/tmp/tmpwMo351/pubring.gpg' created
[bootstrap] gpg: requesting key 914D5813 from hkp server keyserver.ubuntu.com
[bootstrap] gpg: /tmp/tmpwMo351/trustdb.gpg: trustdb created
[bootstrap] gpg: key 914D5813: public key "Launchpad PPA for Government Digital Service" imported
[bootstrap] gpg: Total number processed: 1
[bootstrap] gpg:               imported: 1  (RSA: 1)
[bootstrap] OK
[bootstrap] You are about to add the following PPA to your system:
[bootstrap]
[bootstrap] More info: https://launchpad.net/~gds/+archive/govuk
[bootstrap] --> Check for Ruby
[bootstrap] dpkg-preconfigure: Unable to re-open stdin: No such file or directory
[bootstrap] Selecting previously unselected package libreadline5.
[bootstrap] (Reading database ... 53796 files and directories currently installed.)
[bootstrap] Unpacking libreadline5 (from .../libreadline5_5.2-11_amd64.deb) ...
[bootstrap] Selecting previously unselected package libyaml-0-2.
[bootstrap] Unpacking libyaml-0-2 (from .../libyaml-0-2_0.1.4-2ubuntu0.12.04.2_amd64.deb) ...
[bootstrap] Selecting previously unselected package libtcmalloc-minimal0.
[bootstrap] Unpacking libtcmalloc-minimal0 (from .../libtcmalloc-minimal0_1.7-1ubuntu1_amd64.deb) ...
[bootstrap] Selecting previously unselected package libunwind7.
[bootstrap] Unpacking libunwind7 (from .../libunwind7_0.99-0.3ubuntu1_amd64.deb) ...
[bootstrap] Selecting previously unselected package libgoogle-perftools0.
[bootstrap] Unpacking libgoogle-perftools0 (from .../libgoogle-perftools0_1.7-1ubuntu1_amd64.deb) ...
[bootstrap] Selecting previously unselected package libgoogle-perftools-dev.
[bootstrap] Unpacking libgoogle-perftools-dev (from .../libgoogle-perftools-dev_1.7-1ubuntu1_amd64.deb) ...
[bootstrap] Selecting previously unselected package libruby1.9.1.
[bootstrap] Unpacking libruby1.9.1 (from .../libruby1.9.1_1%3a1.9.3.194-0~51~precise1_amd64.deb) ...
[bootstrap] Selecting previously unselected package ruby1.9.1.
[bootstrap] Unpacking ruby1.9.1 (from .../ruby1.9.1_1%3a1.9.3.194-0~51~precise1_amd64.deb) ...
[bootstrap] Selecting previously unselected package ruby1.9.1-dev.
[bootstrap] Unpacking ruby1.9.1-dev (from .../ruby1.9.1-dev_1%3a1.9.3.194-0~51~precise1_amd64.deb) ...
[bootstrap] Processing triggers for man-db ...
[bootstrap] Setting up libreadline5 (5.2-11) ...
[bootstrap] Setting up libyaml-0-2 (0.1.4-2ubuntu0.12.04.2) ...
[bootstrap] Setting up libtcmalloc-minimal0 (1.7-1ubuntu1) ...
[bootstrap] Setting up libunwind7 (0.99-0.3ubuntu1) ...
[bootstrap] Setting up libgoogle-perftools0 (1.7-1ubuntu1) ...
[bootstrap] Setting up libgoogle-perftools-dev (1.7-1ubuntu1) ...
[bootstrap] Setting up libruby1.9.1 (1:1.9.3.194-0~51~precise1) ...
[bootstrap] Setting up ruby1.9.1 (1:1.9.3.194-0~51~precise1) ...
[bootstrap] update-alternatives: using /usr/bin/gem1.9.1 to provide /usr/bin/gem (gem) in auto mode.
[bootstrap] update-alternatives: using /usr/bin/ruby1.9.1 to provide /usr/bin/ruby (ruby) in auto mode.
[bootstrap] Setting up ruby1.9.1-dev (1:1.9.3.194-0~51~precise1) ...
[bootstrap] Processing triggers for libc-bin ...
[bootstrap] ldconfig deferred processing now taking place
[bootstrap] --> Check for Bundler
[bootstrap] Successfully installed bundler-1.5.3
[bootstrap] 1 gem installed
[bootstrap] --> Bundle gems
[bootstrap] --> Update 3rd-party modules
/var/lib/gems/1.9.1/gems/bundler-1.5.3/lib/bundler/runtime.rb:220: warning: Insecure world writable dir /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/bin in PATH
, mode 040777
/var/lib/gems/1.9.1/gems/bundler-1.5.3/lib/bundler/runtime.rb:220: warning: Insecure world writable dir /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/bin in PATH
, mode 040777
Error: Operation not permitted - /var/apps/pp-puppet/.tmp/librarian/cache/source/puppet/forge/3792e516e3ff92a0ef9f5e827f8e76eb/puppetlabs/stdlib/version/804879a
f3577ffc462dba2bf74cceb6b/stdlib/spec/unit/puppet/parser/functions/is_domain_name_spec.rb
Error: Try 'puppet help module install' for usage
/var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-puppet-0.9.13/lib/librarian/puppet/source/forge.rb:114:in `unlink': Directory not empty - /var/apps/
pp-puppet/.tmp/librarian/cache/source/puppet/forge/3792e516e3ff92a0ef9f5e827f8e76eb/puppetlabs/stdlib/version/804879af3577ffc462dba2bf74cceb6b (Errno::ENOTEMPTY
)
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-puppet-0.9.13/lib/librarian/puppet/source/forge.rb:114:in `unlink'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-puppet-0.9.13/lib/librarian/puppet/source/forge.rb:114:in `cache_version_unpacked!'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-puppet-0.9.13/lib/librarian/puppet/source/forge.rb:57:in `install_version!'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-puppet-0.9.13/lib/librarian/puppet/source/forge.rb:263:in `install!'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-0.1.2/lib/librarian/manifest.rb:101:in `install!'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-0.1.2/lib/librarian/action/install.rb:49:in `block in install_manifests'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-0.1.2/lib/librarian/action/install.rb:48:in `each'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-0.1.2/lib/librarian/action/install.rb:48:in `install_manifests'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-0.1.2/lib/librarian/action/install.rb:39:in `perform_installation'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-0.1.2/lib/librarian/action/install.rb:12:in `run'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-0.1.2/lib/librarian/cli.rb:165:in `install!'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-puppet-0.9.13/lib/librarian/puppet/cli.rb:71:in `install'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/thor-0.18.1/lib/thor/command.rb:27:in `run'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/thor-0.18.1/lib/thor/invocation.rb:120:in `invoke_command'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/thor-0.18.1/lib/thor.rb:363:in `dispatch'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/thor-0.18.1/lib/thor/base.rb:439:in `start'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-0.1.2/lib/librarian/cli.rb:26:in `block (2 levels) in bin!'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-0.1.2/lib/librarian/cli.rb:31:in `returning_status'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-0.1.2/lib/librarian/cli.rb:26:in `block in bin!'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-0.1.2/lib/librarian/cli.rb:47:in `with_environment'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-0.1.2/lib/librarian/cli.rb:26:in `bin!'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-puppet-0.9.13/bin/librarian-puppet:7:in `<top (required)>'
        from ./bin/librarian-puppet:12:in `load'
        from ./bin/librarian-puppet:12:in `<main>'
The following SSH command responded with a non-zero exit status.
Vagrant assumes that this means the command failed!

chmod +x /tmp/vagrant-shell && /tmp/vagrant-shell

Stdout from the command:

[bootstrap] --> Set root profile to quieten sudo tty message
[bootstrap] --> Set nameserver to 8.8.8.8
[bootstrap] --> Set sources.list to gb and update apt cache
[bootstrap] dpkg-preconfigure: Unable to re-open stdin: No such file or directory
[bootstrap] Selecting previously unselected package libcurl3.
[bootstrap] (Reading database ... 53042 files and directories currently installed.)
[bootstrap] Unpacking libcurl3 (from .../libcurl3_7.22.0-3ubuntu4.7_amd64.deb) ...
[bootstrap] Selecting previously unselected package curl.
[bootstrap] Unpacking curl (from .../curl_7.22.0-3ubuntu4.7_amd64.deb) ...
[bootstrap] Selecting previously unselected package liberror-perl.
[bootstrap] Unpacking liberror-perl (from .../liberror-perl_0.17-1_all.deb) ...
[bootstrap] Selecting previously unselected package git-man.
[bootstrap] Unpacking git-man (from .../git-man_1%3a1.7.9.5-1_all.deb) ...
[bootstrap] Selecting previously unselected package git.
[bootstrap] Unpacking git (from .../git_1%3a1.7.9.5-1_amd64.deb) ...
[bootstrap] Selecting previously unselected package unattended-upgrades.
[bootstrap] Unpacking unattended-upgrades (from .../unattended-upgrades_0.76ubuntu1_all.deb) ...
[bootstrap] Selecting previously unselected package python-pycurl.
[bootstrap] Unpacking python-pycurl (from .../python-pycurl_7.19.0-4ubuntu3_amd64.deb) ...
[bootstrap] Selecting previously unselected package python-software-properties.
[bootstrap] Unpacking python-software-properties (from .../python-software-properties_0.82.7.7_all.deb) ...
[bootstrap] Processing triggers for man-db ...
[bootstrap] Processing triggers for ureadahead ...
[bootstrap] Setting up libcurl3 (7.22.0-3ubuntu4.7) ...
[bootstrap] Setting up curl (7.22.0-3ubuntu4.7) ...
[bootstrap] Setting up liberror-perl (0.17-1) ...
[bootstrap] Setting up git-man (1:1.7.9.5-1) ...
[bootstrap] Setting up git (1:1.7.9.5-1) ...
[bootstrap] Setting up unattended-upgrades (0.76ubuntu1) ...
[bootstrap] Setting up python-pycurl (7.19.0-4ubuntu3) ...
[bootstrap] Setting up python-software-properties (0.82.7.7) ...
[bootstrap] Processing triggers for libc-bin ...
[bootstrap] ldconfig deferred processing now taking place
[bootstrap] gpg: keyring `/tmp/tmpwMo351/secring.gpg' created
[bootstrap] gpg: keyring `/tmp/tmpwMo351/pubring.gpg' created
[bootstrap] gpg: requesting key 914D5813 from hkp server keyserver.ubuntu.com
[bootstrap] gpg: /tmp/tmpwMo351/trustdb.gpg: trustdb created
[bootstrap] gpg: key 914D5813: public key "Launchpad PPA for Government Digital Service" imported
[bootstrap] gpg: Total number processed: 1
[bootstrap] gpg:               imported: 1  (RSA: 1)
[bootstrap] OK
[bootstrap] You are about to add the following PPA to your system:
[bootstrap]
[bootstrap] More info: https://launchpad.net/~gds/+archive/govuk
[bootstrap] --> Check for Ruby
[bootstrap] dpkg-preconfigure: Unable to re-open stdin: No such file or directory
[bootstrap] Selecting previously unselected package libreadline5.
[bootstrap] (Reading database ... 53796 files and directories currently installed.)
[bootstrap] Unpacking libreadline5 (from .../libreadline5_5.2-11_amd64.deb) ...
[bootstrap] Selecting previously unselected package libyaml-0-2.
[bootstrap] Unpacking libyaml-0-2 (from .../libyaml-0-2_0.1.4-2ubuntu0.12.04.2_amd64.deb) ...
[bootstrap] Selecting previously unselected package libtcmalloc-minimal0.
[bootstrap] Unpacking libtcmalloc-minimal0 (from .../libtcmalloc-minimal0_1.7-1ubuntu1_amd64.deb) ...
[bootstrap] Selecting previously unselected package libunwind7.
[bootstrap] Unpacking libunwind7 (from .../libunwind7_0.99-0.3ubuntu1_amd64.deb) ...
[bootstrap] Selecting previously unselected package libgoogle-perftools0.
[bootstrap] Unpacking libgoogle-perftools0 (from .../libgoogle-perftools0_1.7-1ubuntu1_amd64.deb) ...
[bootstrap] Selecting previously unselected package libgoogle-perftools-dev.
[bootstrap] Unpacking libgoogle-perftools-dev (from .../libgoogle-perftools-dev_1.7-1ubuntu1_amd64.deb) ...
[bootstrap] Selecting previously unselected package libruby1.9.1.
[bootstrap] Unpacking libruby1.9.1 (from .../libruby1.9.1_1%3a1.9.3.194-0~51~precise1_amd64.deb) ...
[bootstrap] Selecting previously unselected package ruby1.9.1.
[bootstrap] Unpacking ruby1.9.1 (from .../ruby1.9.1_1%3a1.9.3.194-0~51~precise1_amd64.deb) ...
[bootstrap] Selecting previously unselected package ruby1.9.1-dev.
[bootstrap] Unpacking ruby1.9.1-dev (from .../ruby1.9.1-dev_1%3a1.9.3.194-0~51~precise1_amd64.deb) ...
[bootstrap] Processing triggers for man-db ...
[bootstrap] Setting up libreadline5 (5.2-11) ...
[bootstrap] Setting up libyaml-0-2 (0.1.4-2ubuntu0.12.04.2) ...
[bootstrap] Setting up libtcmalloc-minimal0 (1.7-1ubuntu1) ...
[bootstrap] Setting up libunwind7 (0.99-0.3ubuntu1) ...
[bootstrap] Setting up libgoogle-perftools0 (1.7-1ubuntu1) ...
[bootstrap] Setting up libgoogle-perftools-dev (1.7-1ubuntu1) ...
[bootstrap] Setting up libruby1.9.1 (1:1.9.3.194-0~51~precise1) ...
[bootstrap] Setting up ruby1.9.1 (1:1.9.3.194-0~51~precise1) ...
[bootstrap] update-alternatives: using /usr/bin/gem1.9.1 to provide /usr/bin/gem (gem) in auto mode.
[bootstrap] update-alternatives: using /usr/bin/ruby1.9.1 to provide /usr/bin/ruby (ruby) in auto mode.
[bootstrap] Setting up ruby1.9.1-dev (1:1.9.3.194-0~51~precise1) ...
[bootstrap] Processing triggers for libc-bin ...
[bootstrap] ldconfig deferred processing now taking place
[bootstrap] --> Check for Bundler
[bootstrap] Successfully installed bundler-1.5.3
[bootstrap] 1 gem installed
[bootstrap] --> Bundle gems
[bootstrap] --> Update 3rd-party modules


Stderr from the command:

stdin: is not a tty
/var/lib/gems/1.9.1/gems/bundler-1.5.3/lib/bundler/runtime.rb:220: warning: Insecure world writable dir /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/bin in PATH
, mode 040777
/var/lib/gems/1.9.1/gems/bundler-1.5.3/lib/bundler/runtime.rb:220: warning: Insecure world writable dir /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/bin in PATH
, mode 040777
Error: Operation not permitted - /var/apps/pp-puppet/.tmp/librarian/cache/source/puppet/forge/3792e516e3ff92a0ef9f5e827f8e76eb/puppetlabs/stdlib/version/804879a
f3577ffc462dba2bf74cceb6b/stdlib/spec/unit/puppet/parser/functions/is_domain_name_spec.rb
Error: Try 'puppet help module install' for usage
/var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-puppet-0.9.13/lib/librarian/puppet/source/forge.rb:114:in `unlink': Directory not empty - /var/apps/
pp-puppet/.tmp/librarian/cache/source/puppet/forge/3792e516e3ff92a0ef9f5e827f8e76eb/puppetlabs/stdlib/version/804879af3577ffc462dba2bf74cceb6b (Errno::ENOTEMPTY
)
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-puppet-0.9.13/lib/librarian/puppet/source/forge.rb:114:in `unlink'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-puppet-0.9.13/lib/librarian/puppet/source/forge.rb:114:in `cache_version_unpacked!'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-puppet-0.9.13/lib/librarian/puppet/source/forge.rb:57:in `install_version!'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-puppet-0.9.13/lib/librarian/puppet/source/forge.rb:263:in `install!'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-0.1.2/lib/librarian/manifest.rb:101:in `install!'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-0.1.2/lib/librarian/action/install.rb:49:in `block in install_manifests'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-0.1.2/lib/librarian/action/install.rb:48:in `each'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-0.1.2/lib/librarian/action/install.rb:48:in `install_manifests'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-0.1.2/lib/librarian/action/install.rb:39:in `perform_installation'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-0.1.2/lib/librarian/action/install.rb:12:in `run'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-0.1.2/lib/librarian/cli.rb:165:in `install!'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-puppet-0.9.13/lib/librarian/puppet/cli.rb:71:in `install'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/thor-0.18.1/lib/thor/command.rb:27:in `run'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/thor-0.18.1/lib/thor/invocation.rb:120:in `invoke_command'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/thor-0.18.1/lib/thor.rb:363:in `dispatch'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/thor-0.18.1/lib/thor/base.rb:439:in `start'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-0.1.2/lib/librarian/cli.rb:26:in `block (2 levels) in bin!'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-0.1.2/lib/librarian/cli.rb:31:in `returning_status'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-0.1.2/lib/librarian/cli.rb:26:in `block in bin!'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-0.1.2/lib/librarian/cli.rb:47:in `with_environment'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-0.1.2/lib/librarian/cli.rb:26:in `bin!'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-puppet-0.9.13/bin/librarian-puppet:7:in `<top (required)>'
        from ./bin/librarian-puppet:12:in `load'
        from ./bin/librarian-puppet:12:in `<main>'


```

**Comment by robyoung** [comment]
Before I go anywhere I'd like to caution this with the fact that we currently do not support Windows and do not have a plan to.

However, on the surface your problem doesn't look like it's Windows related. Puppet is failing while trying to install `stdlib`.

You can see, first bundler throws some warnings about a directory being world writable:

``` shell
/var/lib/gems/1.9.1/gems/bundler-1.5.3/lib/bundler/runtime.rb:220: warning: Insecure world writable dir /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/bin in PATH
, mode 040777
/var/lib/gems/1.9.1/gems/bundler-1.5.3/lib/bundler/runtime.rb:220: warning: Insecure world writable dir /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/bin in PATH
, mode 040777
```

And then fails trying to install the `stdlib` puppet module:

``` shell
Error: Operation not permitted - /var/apps/pp-puppet/.tmp/librarian/cache/source/puppet/forge/3792e516e3ff92a0ef9f5e827f8e76eb/puppetlabs/stdlib/version/804879a
f3577ffc462dba2bf74cceb6b/stdlib/spec/unit/puppet/parser/functions/is_domain_name_spec.rb
Error: Try 'puppet help module install' for usage
```

The error is due to a temporary directory that it works with being left around with some stuff in it. Basically, the puppet module has become corrupt during the install. We have seen this error a few times before but have not been able to completely prevent it yet. We have a script in [pp-puppet](https://github.com/alphagov/pp-puppet) that hard reinstalls puppet's dependencies. You can run it by sshing onto the virtual machine and then running the following.

``` shell
cd /var/apps/pp-puppet
./tools/reinstall_dependencies
```

Could you run that and let us know what happens please?

**Comment by IanHopkinson** [comment]
OK - thanks for getting back so quickly!

I had to run dos2unix on reinstall_dependencies for it to run. (Paul Furley is about 6 feet from me and helped out!).

This successfully installs a bunch of ruby gems but then fails with the message below.

I'll be working with my colleague, Peter Waller, on this and he is Ubuntu-based so this is non-critical. That said, I'm happy to continue trying things out.

```
+ popd
/var/apps/pp-puppet
+ install_puppet_modules
+ echo 'Fresh installing puppet modules.'
Fresh installing puppet modules.
+ pushd ./tools/..
/var/apps/pp-puppet /var/apps/pp-puppet
+ bundle exec librarian-puppet install
/var/lib/gems/1.9.1/gems/bundler-1.5.3/lib/bundler/runtime.rb:220: warning: Insecure world writable dir /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/bin in PATH
, mode 040777
/var/lib/gems/1.9.1/gems/bundler-1.5.3/lib/bundler/runtime.rb:220: warning: Insecure world writable dir /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/bin in PATH
, mode 040777
Error: Operation not permitted - /var/apps/pp-puppet/.tmp/librarian/cache/source/puppet/forge/3792e516e3ff92a0ef9f5e827f8e76eb/puppetlabs/stdlib/version/804879a
f3577ffc462dba2bf74cceb6b/stdlib/spec/unit/puppet/parser/functions/is_domain_name_spec.rb
Error: Try 'puppet help module install' for usage
/var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-puppet-0.9.13/lib/librarian/puppet/source/forge.rb:114:in `unlink': Directory not empty - /var/apps/
pp-puppet/.tmp/librarian/cache/source/puppet/forge/3792e516e3ff92a0ef9f5e827f8e76eb/puppetlabs/stdlib/version/804879af3577ffc462dba2bf74cceb6b (Errno::ENOTEMPTY
)
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-puppet-0.9.13/lib/librarian/puppet/source/forge.rb:114:in `unlink'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-puppet-0.9.13/lib/librarian/puppet/source/forge.rb:114:in `cache_version_unpacked!'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-puppet-0.9.13/lib/librarian/puppet/source/forge.rb:57:in `install_version!'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-puppet-0.9.13/lib/librarian/puppet/source/forge.rb:263:in `install!'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-0.1.2/lib/librarian/manifest.rb:101:in `install!'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-0.1.2/lib/librarian/action/install.rb:49:in `block in install_manifests'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-0.1.2/lib/librarian/action/install.rb:48:in `each'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-0.1.2/lib/librarian/action/install.rb:48:in `install_manifests'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-0.1.2/lib/librarian/action/install.rb:39:in `perform_installation'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-0.1.2/lib/librarian/action/install.rb:12:in `run'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-0.1.2/lib/librarian/cli.rb:165:in `install!'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-puppet-0.9.13/lib/librarian/puppet/cli.rb:71:in `install'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/thor-0.18.1/lib/thor/command.rb:27:in `run'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/thor-0.18.1/lib/thor/invocation.rb:120:in `invoke_command'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/thor-0.18.1/lib/thor.rb:363:in `dispatch'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/thor-0.18.1/lib/thor/base.rb:439:in `start'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-0.1.2/lib/librarian/cli.rb:26:in `block (2 levels) in bin!'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-0.1.2/lib/librarian/cli.rb:31:in `returning_status'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-0.1.2/lib/librarian/cli.rb:26:in `block in bin!'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-0.1.2/lib/librarian/cli.rb:47:in `with_environment'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-0.1.2/lib/librarian/cli.rb:26:in `bin!'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-puppet-0.9.13/bin/librarian-puppet:7:in `<top (required)>'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/bin/librarian-puppet:23:in `load'
        from /var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/bin/librarian-puppet:23:in `<main>'
```

**Comment by robyoung** [comment]
Could you run that script from on the virtual machine? The idea is to work entirely within a well known environment.
- SSH onto the virtual machine
- `cd /var/apps/pp-puppet`
- `./tools/install_dependencies`

**Comment by IanHopkinson** [comment]
I ran the `reinstall_dependencies` command previously having ssh'd into the virtual machine. The virtual machine does not have an `install_dependencies` script, only the `reinstall_dependencies` script.

**Comment by robyoung** [comment]
Sorry, that was a typo and a misunderstanding all wrapped up into one.

That is strange that you had to run dos2unix on reinstall_dependencies. I do not have windows line endings here. Would be good to run through the install with you when you're in the office and see if we can spot what's going on.

**Comment by IanHopkinson** [comment]
Yes, let's do that. My understanding from Paul was that the install files were shared between linux VM and Windows which may be why they picked up Windows style file endings.

There are subtleties with Windows line endings and GitHub, see here:

https://help.github.com/articles/dealing-with-line-endings

**Comment by pwaller** [comment]
Note that when I try to `vagrant up` I get the following error if I don't install nfs packages, which seems to imply that it won't work on windows:

```
There are errors in the configuration of this machine. Please fix
the following errors and try again:

vm:
* It appears your machine doesn't support NFS, or there is not an
adapter to enable NFS on this machine for Vagrant. Please verify
that `nfsd` is installed on your machine, and try again. If you're
on Windows, NFS isn't supported. If the problem persists, please
contact Vagrant support.
```

**Comment by easternbloc** [comment]
@pwaller take a look at this commit. Try changing the Vagrantfile to look like this and run it again.

https://github.com/alphagov/pp-development/commit/a4e6879d2f7d3bf281c491cf18e9213850082b4c

**Comment by pwaller** [comment]
sorry @easternbloc, I was just noting that it looked like this would fail anyway on windows machines. (I'm not on a windows machine)

Also, were you meaning to remove `:nfs => true`, or add it? It's already in my configuration. I am actually having trouble with /var/apps, it doesn't appear to be mounted..

**Comment by easternbloc** [comment]
I was suggesting removing the nfs option for windows. This should be OK on a mac and *nix.

If you're having issues you might want to remove it and see if /var/apps gets mounted though...

**Comment by pwaller** [comment]
I'm getting the same original error as @IanHopkinson now on my Ubuntu machine. I've had to disable NFS since it isn't supported by the filesystem used by my home directory.

```
Error: No such file or directory - /var/apps/pp-puppet/.tmp/librarian/cache/source/puppet/forge/3792e516e3ff92a0ef9f5e827f8e76eb/smarchive/archive/version/7663d0c47292d3c50eb71d008ed8a340/archive/spec/fixtures/modules/archive/files
Error: Try 'puppet help module install' for usage
/var/apps/pp-puppet/vendor/bundle/ruby/1.9.1/gems/librarian-puppet-0.9.13/lib/librarian/puppet/source/forge.rb:114:in `unlink': Directory not empty - /var/apps/pp-puppet/.tmp/librarian/cache/source/puppet/forge/3792e516e3ff92a0ef9f5e827f8e76eb/smarchive/archive/version/7663d0c47292d3c50eb71d008ed8a340 (Errno::ENOTEMPTY)
```

**Comment by pwaller** [comment]
This appears to [be a puppet bug](http://projects.puppetlabs.com/issues/17033), afaict.

**Comment by pwaller** [comment]
From reading around on the issue a bit and talking to @easternbloc it seems as though if we attempt it a second time it may succeed. It has certainly gotten a lot further than the first time.

**Comment by pwaller** [comment]
Provisioning finally succeeded. Since it looks like an expensive issue to fix fundamentally I've made a pull request to add a note to retry the provisioning in #40.
